components/ephemeral_storage/browser/ephemeral_storage_browsertest.cc

```diff
@@ -0,0 +1,164 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/net/brave_network_delegate_browsertest.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "content/public/browser/notification_types.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/test_navigation_observer.h"
+
+using content::RenderFrameHost;
+
+namespace {
+
+enum StorageType { Session, Local };
+
+void SetStorageValueInFrame(RenderFrameHost* host,
+                            std::string key,
+                            std::string value,
+                            StorageType storage_type) {
+  std::string script = base::StringPrintf(
+      "%sStorage.setItem('%s', '%s');",
+      storage_type == StorageType::Session ? "session" : "local", key.c_str(),
+      value.c_str());
+  ASSERT_TRUE(content::ExecuteScript(host, script));
+}
+
+content::EvalJsResult GetStorageValueInFrame(RenderFrameHost* host,
+                                             std::string key,
+                                             StorageType storage_type) {
+  std::string script = base::StringPrintf(
+      "%sStorage.getItem('%s');",
+      storage_type == StorageType::Session ? "session" : "local", key.c_str());
+  return content::EvalJs(host, script);
+}
+
+bool NavigateRenderFrameToURL(content::RenderFrameHost* frame,
+                              std::string iframe_id,
+                              const GURL& url) {
+  std::string script = base::StringPrintf(
+      "setTimeout(\""
+      "var iframes = document.getElementById('%s');iframes.src='%s';"
+      "\",0)",
+      iframe_id.c_str(), url.spec().c_str());
+
+  content::TestNavigationManager navigation_manager(
+      content::WebContents::FromRenderFrameHost(frame), url);
+  bool result = ExecuteScript(frame, script);
+  navigation_manager.WaitForNavigationFinished();
+  return result;
+}
+
+}  // namespace
+
+class EphemeralStorageBrowserTest : public BraveNetworkDelegateBrowserTest {
+ public:
+  EphemeralStorageBrowserTest() : BraveNetworkDelegateBrowserTest() {}
+
+  void SetUpOnMainThread() override {
+    BraveNetworkDelegateBrowserTest::SetUpOnMainThread();
+    ephemeral_storage_url_ =
+        https_server_.GetURL("a.com", "/ephemeral_storage.html");
+    simple_url_ = https_server_.GetURL("a.com", "/simple.html");
+  }
+
+ protected:
+  GURL ephemeral_storage_url_;
+  GURL simple_url_;
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(EphemeralStorageBrowserTest);
+};
+
+IN_PROC_BROWSER_TEST_F(EphemeralStorageBrowserTest,
```

Comment:
I'm confused, I don't see where you've enabled kBraveEphemeralStorage so isn't this testing the default behavior?

Reply:
these tests (and an existing test (LocalStorageTest) are all failing in ci

Reply:
The test needs to be updated to work with the runtime feature, which is something I uploaded to #6615. If you don't mind reviewing that, I can work on getting this past the CI today.